app/views/site/_study_visualize.html.erb

```diff
@@ -1,52 +1,82 @@
-                <div class="panel-body">
-                </div>
-              </div>
-            </div>
-          </div>
-        <% end %>
-        <% if @study.can_visualize_genome_data? %>
-          <div class="tab-pane<%= !@study.can_visualize_clusters? ? ' active' : nil%>" id="genome-tab" role="tabpanel" >
-            <div class="panel panel-default no-top-border">
-              <div id="genome-container" class="panel-collapse collapse in">
-                <div style="padding: 15px 30px;">
-                  <%= render partial: '/site/genome/genome' %>
-                </div>
-              </div>
-            </div>
-          </div>
-        <% end %>
-      </div>
-    </div>
-  </div>
-</div>
 <script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
   window.SCP.studyAccession = '<%= @study.accession %>';
-  <% if User.feature_flag_for_instance(current_user, 'react_explore')  %>
-    window.SCP.renderExploreView(document.getElementById('study-visualize'), window.SCP.studyAccession)
-  <% else %>
-    window.SCP.exploreDefault();
-  <% end %>
+  window.SCP.renderExploreView(document.getElementById('study-visualize'), window.SCP.studyAccession)
 </script>
+
+
+<% if @study.has_analysis_outputs?('infercnv', 'ideogram.js') or @study.has_bam_files? %>
+  <!--
+  TODO (SCP-3258): Migrate genome visualization scaffold from ERB/Rails to API/React
+
+  Minimal DOM scaffolding and server-side data for genomic visualizations.
+  See partials referenced below and scp-igv.js for larger bodies of code.
+  -->
+  <script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
+    if (typeof accessToken === 'undefined') {
+      window.accessToken = '<%= get_read_access_token(@study, current_user) %>';
+    }
+  </script>
+  <% if @study.has_analysis_outputs?('infercnv', 'ideogram.js') and action_name == 'study' %>
+    <div id="ideogram-container" style="margin-top: 12px">
+      <div id="filters-container">
+        <ul id="tracks-to-display">
+        </ul>
+      </div>
+      <!-- Ideogram goes here -->
+    </div>
+    <style>
+      /** TODO: Migrate this to SCSS as part of SCP-3258 **/
+      #tracks-to-display {
+        list-style-type: none;
+        float: left;
+        padding-left: 20px;
+      }
+
+      #tracks-to-display label {
+        font-weight: normal;
+        position: relative;
+        top: 1px;
+      }
+
+      #tracks-to-display input {
+        margin-right: 5px;
+      }
+
+      #_ideogramLegend {
+        position: relative;
+        left: 20px;
+      }
+
+      #ideogramTitle {
+        margin-left: 110px;
+      }
+    </style>
+    <script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
+      // Intercept requests to add bearer token, enabling direct load of files from GCS
+      // This was needed to load genome visualization data from GCS.
+      //
+      // TODO (SCP-2451): Remove monkey patching of fetch
```

Comment:
How does this intersect with Jon's open PR about switching to pet service accounts for GCS file streaming?

Reply:
I think this work just misses intersection with PR #972, at least right now.  That PR is isolated to refining logic that sets this `accessToken` ([via `get_read_access_token`](https://github.com/broadinstitute/single_cell_portal_core/pull/972/files#diff-5f3fc5e3977d242572aa1d08551f5eb557de0ccaff30370838ee9df5386ea0daR349)), whereas this part of this PR simply inlines code that had been burrowed in another ERB template (setting up SCP-3258, the TODO in the block above this).

To streamline SCP-3258 and avoid toe-stepping between this PR and that PR, it might make sense to update its ACs to include [one of your suggestions](https://github.com/broadinstitute/single_cell_portal_core/pull/972#discussion_r612562040) in @bistline's PR.

Reply:
I am actually concerned about the intersection - based off of some of the comments from #972, I believe that there will now be merge conflicts once I've finished what has been proposed (specifically surrounding `window.accessToken` and the deletion in this PR of `app/views/site/genome/_genome.html.erb`).

We should discuss how we would like to handle this.

Reply:
Yup, the last sentence in my previous comment suggests a way forward.

Reply:
After a brief huddle, we concluded (as Jon notes) that a brief merge conflict seems unavoidable.  Jon plans to resolve it while implementing Devon's suggestion over in #972 -- and to do so in a way that doesn't break this code, which I'll merge now.